### ISSUE TITLE
When I export logits in VLLM during first forward it's more slowly than torch. Does anyone have experience about llama-70B-chat-hf VLLM inference speed with 2 batch size? 

### ISSUE THREAD

**Issue by starlitsky2010** [vllm-project/vllm#1398]
For harness test framework. it uses "logits"to calculate the PPL like this:
```
    def _model_call(
        self, inputs: TokenSequence, labels: Optional[TokenSequence] = None
    ) -> TokenSequence:
        return self.model(inputs)["logits"]
```

Then, I've added the VLLM implement in the test pipeline with exporting the logits for the first forward.

```
def _model_call(
            self, inputs: TokenSequence, labels: Optional[TokenSequence] = None
        ) -> TokenSequence:
        sampling_params = SamplingParams(max_tokens=self._max_gen_toks,
                                         temperature=0.0,
                                         top_p=1.0,
                                         top_k=-1,
                                         logprobs=1,
                                         export_logits=True,
                                         step_limit=1,)  # I added this for only executing one time forward
        prompt_token_ids_list = [input_token[0].squeeze(0).tolist() for input_token in inputs]
        outputs = self.llm.generate(prompt_token_ids=prompt_token_ids_list,
                                    sampling_params=sampling_params,
                                    use_tqdm=False)
        return outputs

```

I found it's running very slowly, when I tested Llama-70B-chat-hf model.

Model: Llama2-chat-70b-hf
VLLM baseline commit ID: bc0644574ca12d754a031596bdcfe8e1f0e6ab39
VLLM batch_size = 2  (In other words, len(prompt_token_ids_list) == 2)
VLLM TP = 8

Using VLLM             inference got test speed  38/38 [03:08<00:00,  4.96s/it] (It equals to 2.48s/it as batch size is 2)
Using huggingface  inference got test speed  76/76 [01:27<00:00,  1.16s/it] (TP=8)

I don't know why VLLM is slowly than huggingface transformers inference engine.
I've tested the time consuming in the pipeline and found 

```
vllm/engine/llm_engine.py
    def _run_workers(
        self,
        method: str,
        *args,
        get_all_outputs: bool = False,
        **kwargs,
    ) -> Any:
......
        if self.parallel_config.worker_use_ray:
            all_outputs = ray.get(all_outputs)
            # ray.get(all_outputs): costs about 7.283660888671875 seconds
......
```

Does anyone have experience about llama-70B-chat-hf VLLM inference speed with 2 batch size? 

Thanks

**Comment by hmellor** [comment]
Closing this issue as stale as there has been no discussion in the past 3 months.

If you are still experiencing the issue you describe, feel free to re-open this issue.
